working/bazel/module-structure.md

```diff
@@ -0,0 +1,234 @@
+The following is a reference for [`Bazel`](https://bazel.build/) (internally
+called `Blaze`) structures modules of Dart code, especially when compared to
+external structures (i.e. via `pub`).
+
+This is *not* a proposal, nor a commitment to build out further Bazel support
+externally, but rather a reference to help guide discussions around visibility
+modifiers and terminology around modular/reusable Dart code.
+
+## Monolothic Repository
+
+In ["Why Google Stores Billions of Lines of Code in a Single Repository"][1]
+the author explain the benefits of a _Monolothic Repository_ ("Mono Repo").
+Many of the constraints described in this document are due to this decision,
+and be expected to be _unchangeable_ - i.e. it is quite outside of the scope
+of any language (including Dart) to change these requirements.
+
+[1]: https://cacm.acm.org/magazines/2016/7/204032-why-google-stores-billions-of-lines-of-code-in-a-single-repository/fulltext
+
```

Comment:
Does this mean that Bazel use is a Google-only requirement that nobody else should worry about? (Again, who is the target audience?)

Reply:
No. Clarified.